pkg/rhos/ocpgwdeployer.go

```diff
@@ -259,6 +277,13 @@ func (d *ocpGatewayDeployer) Cleanup(reporter api.Reporter) error {
 		// If the instance name matches with d.InfraID + "-submariner-gw-", it implies that
 		// the gateway node was deployed using the OCPMachineSet API otherwise it's an existing worker node.
 		prefix := d.InfraID + "-submariner-gw-"
+
+		err = d.removeGWFirewallRules(groupName, gwNodes[i].Name, computeClient)
```

Comment:
```suggestion
		err = d.removeFirewallRulesFromGW(groupName, gwNodes[i].Name, computeClient)
```
Current name seems to suggest we're deleting the SG itself, while we're just removing association from gw node.